src/uu/split/src/split.rs

```diff
@@ -65,6 +67,82 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
     }
 }
 
+/// If the same separator (with the same value) was used multiple times - `split` should NOT fail
+/// If the separator was used multiple times but with different values - `split` should fail
+/// Unfortunately Clap cannot support both conditions simultaneously, so it is setup to fail on the second condition
+/// For the first condition - this function will find if there are multiple separator options with the same value
+/// and remove duplicates, leaving only one
+/// i.e. `split -t: -t:` will become `split -t:`
+/// However it will leave other values untouched, so
+/// `split -t: -t: -t,` will become `split -t: -t,` and Clap will panic next as it should
```

Comment:
Can't we still use clap here with some custom post-processing?

I'm thinking that it should be fairly simple to collect multiple values for the separator and then check that all given values are the same.

Would that give a different result?

The problem with an approach like this one is that you theoretically need to keep _all_ options into account. For example what if the `-t` is the argument to the `--filter` flag. It's unlikely, but possible and would be difficult to fix. Two separate pre-processing steps is also just very messy to deal with. Whenever we get a bad output from `clap` now, it could be clap, `handle_obsolete` or this function that messed something up.

Reply:
yes, we could use Clap's `overrides_with()` pointed to itself - in that case no matter how many times `-t` and/or `--separator` is used - the last one would win. However, that will break GNU test as it checks for both cases mentioned in the comment section for this function, i.e. this specifically will NOT cause Clap to fail with `overrides_with()` pointed to itself, but GNU version expects it to fail:
```
split -ta -tb </dev/null
```

Reply:
> The problem with an approach like this one is that you theoretically need to keep _all_ options into account. For example what if the `-t` is the argument to the `--filter` flag. It's unlikely, but possible and would be difficult to fix. Two separate pre-processing steps is also just very messy to deal with. Whenever we get a bad output from `clap` now, it could be clap, `handle_obsolete` or this function that messed something up.

In this example - it will not break as this function will only remove arg slices for complete duplicates of separator/value pairs (concatenated or not), i.e. in case of `split --filter -t in` nothing will be altered.

Reply:
Ah sorry, I didn't mean `overrides`, but just collecting multiple values, for example with [`ArgAction::append`](https://docs.rs/clap/latest/clap/enum.ArgAction.html#variant.Append). You'd get a `Vec` of all the values that you can check.

Reply:
fantastic suggestion, thank you. Looks like this will work and with less code too :)